src/black/lines.py

```diff
@@ -718,7 +719,7 @@ def is_line_short_enough(line: Line, *, line_length: int, line_str: str = "") ->
     if not line_str:
         line_str = line_to_string(line)
     return (
-        len(line_str) <= line_length
+        str_width(line_str) <= line_length
```

Comment:
This change appears to affect the stable style. It must be made in the preview style only. (The trans.py code is already preview-only so no need to change anything there.)

Reply:
Would it be a good idea to let `is_line_short_enough()` to take `mode: Mode` instead of `line_length: int`?  Or would it be better to just add a new parameter `preview: bool` to the function?

Reply:
I just added a new keyword parameter `preview: bool`.  Please let me know if it's not appropriate.